src/managed/Microsoft.Extensions.DependencyModel/Microsoft.Extensions.DependencyModel.csproj

```diff
@@ -5,30 +5,52 @@
     <Description>Abstractions for reading `.deps` files.</Description>
     <TargetFrameworks>net451;netstandard1.3;netstandard1.6;netstandard2.0</TargetFrameworks>
     <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard1.3;netstandard1.6;netstandard2.0</TargetFrameworks>
+    <LangVersion>7.3</LangVersion>
   </PropertyGroup>
 
   <Choose>
     <!--
-    Newtonsoft.Json v11.0.1 is the first verison that targets netstandard2.0. Since we added a target for netstandard2.0
-    so users aren't forced to download the 1.x dependencies, it makes sense to use this version of Newtonsoft.Json.
-    However, we still use the previous (v9.0.1) for other TFMs, so existing users don't need to upgrade their Newtonsoft.Json.
+    Since we added a target for netstandard2.0 so users aren't forced to download the 1.x dependencies,
+    or depend on Newtonsoft.Json, use the Microsoft.Bcl.Json.Sources package (which targets netstandard2.0).
+    However, we still use the Newtonsoft.Json (v9.0.1) for other TFMs, so existing users don't need to upgrade their Newtonsoft.Json.
     For example, the SDK targets net4x and is loaded in VS, so it can't upgrade to a new Newtonsoft.Json.
     -->
     <When Condition="'$(TargetFramework)' == 'netstandard2.0'">
+      <!--These properties are necessary to build the source package-->
       <PropertyGroup>
-        <NewtonsoftJsonPackageVersion>11.0.1</NewtonsoftJsonPackageVersion>
+        <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+        <!-- Suppress warnings for unnecessary CLSCompliance attributes -->
+        <NoWarn>3021</NoWarn>
       </PropertyGroup>
+      <ItemGroup>
+        <PackageReference Include="Microsoft.Bcl.Json.Sources" Version="4.6.0-preview.19072.2">
+          <PrivateAssets>All</PrivateAssets>
+        </PackageReference>
+        <PackageReference Include="System.Memory" Version="4.5.2" />
```

Comment:
Do we really need `4.5.2`?

Reply:
As opposed to what? 4.5.2 is the latest version available.

Reply:
Right, but is the latest version the right thing to choose? Typically libraries don't update their dependencies just because there is a new version out. Doing so causes anyone who references you to use the latest version, which may not be the right thing.

Is there a specific reason you choose `4.5.2`?  or was it just because it was the latest version?

---
In reply to: [251573648](https://github.com/dotnet/core-setup/pull/5009#discussion_r251573648) [](ancestors = 251573648)

Reply:
> Typically libraries don't update their dependencies just because there is a new version out. Doing so causes anyone who references you to use the latest version, which may not be the right thing.

Good point. I didn't have a specific need for the latest. I believe I can safely downgrade to `4.5.0`.